app/src/main/java/ai/elimu/vitabu/ui/storybook/StoryBookActivity.kt

```diff
@@ -47,6 +51,34 @@ class StoryBookActivity : AppCompatActivity() {
 
         val zoomOutPageTransformer = ZoomOutPageTransformer()
         binding.viewPager.setPageTransformer(true, zoomOutPageTransformer)
+        binding.viewPager.setOnPageChangeListener(object : ViewPager.OnPageChangeListener {
+
+            override fun onPageScrolled(
+                position: Int,
+                positionOffset: Float,
+                positionOffsetPixels: Int
+            ) {
+            }
+
+            override fun onPageSelected(position: Int) {
+                if (position == storyBookChapters.size - 1 && !hasReportedCompletion) {
+                    val storyBookGson = ContentProviderUtil.getStoryBookGson(
```

Comment:
@tuancoltech Since this activity is for reading a storybook that has already been loaded, isn't a `StoryBookGson` instance already available in memory somewhere? (To avoid loading it a second time here.)